### PR TITLE
Add tool to clear files in error.

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        $dataset = $this->entityManager->getRepository(Dataset::class)->findOneBy(
+            array('udi' => $udi)
+        );
+
+        $dql = "SELECT f FROM \App\Entity\File f
+            WHERE f.id = :fileId
+            AND f.fileset = :filesetId
+            AND f.status = :fileStatus";
+
+        $query = $this->entityManager->createQuery($dql);
```
Comment (mickel1138) [inline src/Command/PelagosClearErrorAndFileCommand.php:88]: Why use DQL? You can use a repository, or traverse down Dataset.

Comment (fightingtexasaggie) [reply src/Command/PelagosClearErrorAndFileCommand.php]: converted to using the repo here

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        $fileToDelete = $query->getResult()[0];
+
+        // This section is just added for extra safety. Don't delete
+        // unless the file is represented in the datastore. Don't delete
+        // the only copy of a file.
+        $dqlSafe = "SELECT f FROM \App\Entity\File f
+            WHERE f.id <> :fileId
+            AND f.fileset = :filesetId
+            AND f.physicalFilePath like :newUdi
+            AND f.physicalFilePath not like :uploadDir";
```
[the remark this answers is not in the document]
Comment (fightingtexasaggie) [reply src/Command/PelagosClearErrorAndFileCommand.php]: Filtering with pattern matching for the like/not like ended up being rather messy. DQL is much more straightforward.

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                $query->setParameter('filesetId', $dataset->getDatasetSubmission()->getFileset());
+                $query->setParameter('fileId', $fileId);
+                $query->setParameter('newUdi', preg_replace('/:/', '.', $udi) . '%');
+                $query->setParameter('uploadDir', $this->homedirPrefix . '/upload/files/%');
+                if (count($query->getResult()) > 0) {
+                    try {
+                        $delId = $fileToDelete->getId();
+                        $this->entityManager->remove($fileToDelete);
+                        $this->entityManager->flush();
+                        unlink($fileToDelete->getPhysicalFilePath());
```
Comment (praneethpr) [inline src/Command/PelagosClearErrorAndFileCommand.php:115]: Are we deleting from Uploads directory or from Data store

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+namespace App\Command;
+
+use App\Entity\Dataset;
+use App\Entity\DatasetSubmission;
+use App\Entity\File;
+use App\Entity\Fileset;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use FFI;
```
Comment (fightingtexasaggie) [inline src/Command/PelagosClearErrorAndFileCommand.php:11]: Typo?

**Review by praneethpr — APPROVED** [review]

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                'id' => $fileId,
+                'status' => File::FILE_ERROR
+            )
+        );
+
+        if ($fileToDelete instanceof File) {
+            try {
+                $delId = $fileToDelete->getId();
+                $this->entityManager->remove($fileToDelete);
+                $this->entityManager->flush();
```
Comment (mickel1138) [inline src/Command/PelagosClearErrorAndFileCommand.php:76]: Flush last just in case the actual delete fails.

**Review by mickel1138 — APPROVED** [review]

**Pull request merged**